src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/ServiceHubContextFactory.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal class ServiceHubContextFactory
+    {
+        private readonly ServiceHubLifetimeManagerFactory _managerFactory;
+
+        public ServiceHubContextFactory(ServiceHubLifetimeManagerFactory managerFactory)
+        {
+            _managerFactory = managerFactory;
+        }
+
+        public async Task<IServiceHubContext> Create(string hubName, ILoggerFactory loggerFactory = null, CancellationToken cancellationToken = default)
+        {
+            var manager = await _managerFactory.Create(hubName, cancellationToken, loggerFactory);
+            var servicesPerHub = new ServiceCollection();
```

Comment:
why the service collection is for each hub? isn't it enough for each service manager?

Reply:
Because `IHubContext<Hub>` is for each hub, and one `ServiceManager` could have multiple `HubContext`

Reply:
yeah I am also confused here, why DI ctor not working ? `public ServiceHubContextFactory(ServiceHubLifetimeManagerFactory managerFactory, IHubContext<Hub> context)`

Reply:
For two reasons: 
1. `IHubContext<Hub>` is not singleton. And its impl `HubContext<>` is an internal class, so we can't do `services.AddTransient(typeof(IHubContext<>),typeof(HubContext<>));`.
2. `HubContext<Hub>` depends on `HubLiftTimeManager<>`, which depends on hub name in our impl.

Reply:
IF so, does `ServiceHubContextFactory(ServiceHubLifetimeManagerFactory managerFactory, IServiceProvider sp)`
-> `sp.GetService<IHubContext<Hub>>()` work?